source/broadband/webpa_notification.c

```diff
@@ -1041,12 +1041,50 @@ static void addNotifyMsgToQueue(NotifyData *notifyData)
 	WalPrint("*****Returned from addNotifyMsgToQueue*****\n");
 }
 
+/*
+ * @brief To handle sync notifications during factory reset
+ */
+void FR_CloudSyncCheck()
+{
+	char *dbCID = NULL;
+	int factory_reset_flag = false;
+	for(int i=1;i<=12;i++)
+	{
+		dbCID = getParameterValue(PARAM_CID);
+		if(dbCID == NULL)
+		{
+			WalError("Unable to get dbCID value in handleNotificationEvents\n");
+			sleep(5);
+		}
+		else if (dbCID != NULL && strcmp(dbCID, "0") == 0) //To check wheter the devie is up after factory reset
+		{
+			WalInfo("factory reset cloud sync is in progress, wait 5sec for process nitification\n");
+			factory_reset_flag = true;
+			sleep(5);
+		}
+		else
+		{
+			break;
+		}
+		WAL_FREE(dbCID);
+	}
+
+	if(dbCID != NULL && (strcmp(dbCID, "0") == 0) && factory_reset_flag == true)
```

Comment:
Instead of factory_reset_flag , better add FR_cloud_sync_completed flag and log based on that flag value